external/common/lapack/FindTargetOpenMP.cmake

```diff
@@ -0,0 +1,85 @@
+# FindTargetOpenMP.cmake
+# ----------------------
+#
+# OpenMP cmake module to wrap FindOpenMP.cmake in a target.
+#
+# This module sets the following variables in your project: ::
+#
+#   TargetOpenMP_FOUND - true if OpenMP found on the system
+#   TargetOpenMP_MESSAGE - status message with OpenMP library path list
+#   OpenMP_CXX_LIB_NAMES - ";"-separated list of CXX libraries for OpenMP (no paths)
+#
+# * this is a hack until we update min cmake (to 3.10?)
+# * it is specific to Psi4 in that:
+#   * we only care about C++ OpenMP
+#   * minimum Intel in 2017, so no need to add logic for lower
+#   * we never use OPENMP_FOUND
+# * this is run by the TargetLAPACK module and tgt::MathOpenMP is supplemented by
+#   math libraries detected there
+#
+# This module defines the following imported targets ::
+#
+#   OpenMP::OpenMP_CXX - if enabled and available for compiler for C++.
+#                        may be found by newer FindOpenMP or constructed here.
+#
+
+set(PN TargetOpenMP)
+
+# 1st precedence - libraries passed in through -DOpenMP_LIBRARIES
+if (OpenMP_LIBRARIES AND OpenMP_FLAGS AND OpenMP_CXX_LIB_NAMES)
+    if (NOT ${PN}_FIND_QUIETLY)
+        message (STATUS "OpenMP detection suppressed.")
+    endif()
+
+    add_library (OpenMP::OpenMP_CXX INTERFACE IMPORTED)
+    set_property (TARGET OpenMP::OpenMP_CXX PROPERTY INTERFACE_COMPILE_OPTIONS ${OpenMP_FLAGS})
+    set_property (TARGET OpenMP::OpenMP_CXX PROPERTY INTERFACE_LINK_LIBRARIES ${OpenMP_LIBRARIES})
+else()
+    # 2nd precedence - target from modern FindOpenMP.cmake
+    find_package (OpenMP QUIET MODULE COMPONENTS CXX)
+
+    if (NOT TARGET OpenMP::OpenMP_CXX)
```

Comment:
Neat way of using CMake 3.10 features _without_ checking for the CMake version! Was this series of commands adapated from the newest CMake `FindOpenMP.cmake` file? If yes, could you add a note linking to that file?

Reply:
Glad to add in a note. Not sure what you mean by "series of commands", though. Certainly the names of variables tried to follow https://github.com/Kitware/CMake/blob/master/Modules/FindOpenMP.cmake#L32-L53 .

Reply:
I meant the CMake code.

Reply:
The multiple-fallback is based on the [`FindTargetLAPACK.cmake`](https://github.com/SebWouters/CheMPS2/blob/master/CMake/FindTargetLAPACK.cmake) and `FindTargetHDF5.cmake` that I've been distributing to the psi4 ecosystem so that they can replace `find_package(LAPACK)` with `find_package(TargetLAPACK)` so as to inject psi4-found math but behind the scenes just fall back to cmake-provided math-finding so they don't need any extra equipment. That pattern has been working stably for a while.